qdrant_client/local/local_collection.py

```diff
@@ -1071,6 +1090,57 @@ def search_groups(
 
         return models.GroupsResult(groups=groups_result)
 
+    def facet(
+        self,
+        key: str,
+        facet_filter: Optional[types.Filter] = None,
+        limit: int = 10,
+    ) -> types.FacetResponse:
+        facet_hits: Dict[types.FacetValue, int] = defaultdict(int)
+
+        mask = self._payload_and_non_deleted_mask(facet_filter)
+
+        for idx, payload in enumerate(self.payload):
+            if not mask[idx]:
+                continue
+
+            if not isinstance(payload, dict):
+                continue
+
+            values = value_by_key(payload, key)
+
+            if values is None:
+                continue
+
+            # Only count the same value for each point once
+            values_set: Set[types.FacetValue] = set()
+
+            # Sanitize to use only valid values
+            for v in values:
+                if not isinstance(v, get_args_subscribed(types.FacetValue)):
+                    continue
+
+                # If values are UUIDs, format with hyphens
+                as_uuid = parse_uuid(v)
+                if as_uuid:
+                    v = str(as_uuid)
+
+                values_set.add(v)
+
+            for v in values_set:
+                facet_hits[v] += 1
+
+        hits = [
+            models.FacetValueHit(value=value, count=count)
+            for value, count in sorted(
+                facet_hits.items(),
+                # order by count descending, then by value ascending
+                key=lambda x: (-x[1], x[0]),
+            )[:limit]
+        ]
```

Comment:
if there is a facet request on a field which is not one of supported type (int, str), e.g. with a boolean, then we'll get an ugly pydantic error like 

```
validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 2 validation errors for FacetValueHit
value.int
  Input should be a valid integer [type=int_type, input_value=True, input_type=bool]
    For further information visit https://errors.pydantic.dev/2.9/v/int_type
value.str
  Input should be a valid string [type=string_type, input_value=True, input_type=bool]
    For further information visit https://errors.pydantic.dev/2.9/v/string_type
```

Reply:
I thought this was already sanitized above. Did you create a failing test?

Reply:
I just realized that bool is a subclass of int, which is why it was getting through `isinstance`. 

I now fixed it by using

```python
if type(v) not in get_args_subscribed(types.FacetValue):
    continue
```